Fix bug in LINKS_LIB

rudihorn opened the pull request:
Currently the makefile produces a bash script which initialises the jslib path to `<something>/lib/lib/<something>` This is fixed by removing `lib/` from the Makefile bit that generates the `./links` script.

For reference the links code that generates the full path: https://github.com/links-lang/links/blob/eaa33ca8c7fb4a7bab85424170379c2250d8ac6c/core/webserver.ml#L314

dhil commented:
Can you please clarify what this patch is supposed to do? I am having a really hard time parsing your patch note:
> Currently the makefile produces a bash script which initialises the jslib path to <something>/lib/lib/<something> This is fixed by removing lib/ from the Makefile bit that generates the ./links script.

rudihorn commented:
It fixes the `Startup error: Runtime dependency `jslib.js' is not loaded.` error I was getting when compiling links and running formlets.

jamescheney commented:
`LINKS_LIB` is the path used to find Links libraries like`mvu.links` which is why this change makes the CI fail. 

`jslibdir` and `jslib` should be configured manually to the same values as in the auto generated config file if you are using a custom configuration.

rudihorn commented:
What I'm currently doing is calling `make` and then running `./links` from the same directory without any changed config. This produces the path `./_build/default/lib/lib/js/jslib.js` for the jslib directory, but this path does not exist. The correct path is ``./_build/default/lib/js/jslib.js``. Maybe this was changed with a new build or something. I'll try running the tests locally as well, maybe try updating dune on the CI.

rudihorn commented:
I've switched it so that the extra `lib` is now removed from `core/webserver.ml` and this also fixes the jslib problem while not breaking the stdlib directory.

So after looking again my new suspicion is that this has been broken for 7 months since module changes happened. It isn't tested by the CI, and I'm assuming everyone who works on links is setting `jslib` rather than leaving it at its default.

rudihorn commented:
I've added an error for if the jslib directory can't be found which also prints out the path it was looking at. Unfortunately I can't test it because if I move or delete `_build/default/lib/js` then I get the error:

> ***: Error: Cannot locate database driver 'postgresql'

jamescheney commented:
> if I move or delete _build/default/lib/js then I get the error:
> 
>    ***: Error: Cannot locate database driver 'postgresql'

I'm not sure why that's happening; it even happens if I move `.../lib/jslib.js` somewhere else without changing the directory structure, and even if I'm not doing anything web related (e.g. running database tests).  That seems like a bug in the loader or configuration since the absence of this file should have nothing to do with whether loading the database driver succeeds.  

Presuambly you can test whether the error check does the right thing by trying a web-only example that doesn't have any database configuration issues.

jamescheney commented on core/webserver.ml:
```diff
@@ -311,10 +311,16 @@ struct
              end
           | Some s -> s
         in
+        if Sys.file_exists linkslib |> not
```
Like this?

```suggestion
        if not (Sys.file_exists linkslib)
```

jamescheney commented:
Can this get merged?  It seems like a small and uncontroversial change (other than the thing about "not").
